### DISCUSSION TITLE
Does tcp_connect_timeout actually applicable?

### DISCUSSION THREAD

**Discussion by PaulWasTaken** [oracle/python-oracledb#48] [General]
Hello, first of all I am very gratefull for this new version of cx_Oracle - oracledb)

Nevetheless I have somethig to ask.
Does `tcp_connect_timeout` actually applicable? I came up with the question after some research. 
I tryied to dive deep into the sources and can not find any place where it possible could be applied.
So, if I use pool, the connecection is made here https://github.com/oracle/python-oracledb/blob/main/src/oracledb/pool.py#L152
based on https://github.com/oracle/python-oracledb/blob/main/src/oracledb/connection.py#L65 and I can not pass conn_params via `acquire`.
I thought that pool params will be used, but I can not find the place.
It is passed here https://github.com/oracle/python-oracledb/blob/main/src/oracledb/connection.py#L137 but it seems not to be used here https://github.com/oracle/python-oracledb/blob/47c3b2cc1d7a5994c36684b762f66a9c56a1466b/src/oracledb/impl/thick/connection.pyx#L255

Could you make some clarification please
Will `tcp_connect_timeout` be used if it is passed in my dsn?

**Comment by anthony-tuininga** [comment]
The parameter `tcp_connect_timeout` is indeed used! In thin mode, you can see it used [here](https://github.com/oracle/python-oracledb/blob/main/src/oracledb/impl/thin/protocol.pyx#L252). In thick mode, the parameter is used internally by the Oracle Client libraries so you will not see it in the code. For this reason it needs to be in the connect string (called TRANSPORT_CONNECT_TIMEOUT there). The connect string will be built for you if you only use parameters. If, however, you use a connect string without that setting and then use the `tcp_connect_timeout` parameter at the same time, it will not work. So, this:

```python
connect(user=USER, password=PASSWORD, dsn="host/service_name", tcp_connect_timeout=30)
```

will not work, whereas this:

```python
connect(user=USER, password=PASSWORD, host="host", service_name="service_name", tcp_connect_timeout=30)
```

will work. Both cases work in thin mode, though.

Hopefully that explains things!

**Reply by PaulWasTaken** [reply]
Hm, I may be misunderstanding, 

> For this reason it needs to be in the connect string

If I
```
pool = oracledb.create_pool(
                user=self._user,
                password=self._password,
                dsn=self._dsn,
                min=self._pool_min_size,
                max=self._pool_max_size,
                timeout=self._session_timeout,
                wait_timeout=self._acquire_timeout,
                tcp_connect_timeout=self._connect_timeout,  <- here
                getmode=oracledb.POOL_GETMODE_TIMEDWAIT,
)
```
and then 
```
with self._db_pool.acquire() as conn:
    ....
```
it won't work, but if I add `f'(TRANSPORT_CONNECT_TIMEOUT={self._connect_timeout})'` to my dsn it will?

**Reply by PaulWasTaken** [reply]
Oh, sorry, I reread your answer especially

> if, however, you use a connect string without that setting and then use the tcp_connect_timeout parameter at the same time, it will not work

Thanks a lot!

**Reply by anthony-tuininga** [reply]
You're welcome! Glad to hear you got it worked out.
